manifests/modules_load.pp

```diff
@@ -0,0 +1,59 @@
+# Creates a modules-load.d drop file
+#
+# @api public
+#
+# @see modules-load.d(5)
+#
+# @param filename
+#   The name of the modules-load.d file to create
+#
+# @param ensure
+#   Whether to drop a file or remove it
+#
+# @param path
+#   The path to the main systemd modules-load.d directory
+#
+# @param content
+#   The literal content to write to the file
+#
+#   * Mutually exclusive with ``$source``
+#
+# @param source
+#   A ``File`` resource compatible ``source``
+#
+#   * Mutually exclusive with ``$content``
+#
+# @example load a module
+#   systemd::modules_load{'impi.conf':
+#      content => "ipmi\n",
+#   }
+#
+# @example override a file in /lib/modules-load.d
+#   systemd::modules_load{'myservice.conf':
+#      content => "# Cancel system version of the file\n",
+#   }
+#
+define systemd::modules_load (
+  Enum['present', 'absent', 'file'] $ensure   = 'file',
+  Systemd::Dropin                   $filename = $name,
+  Stdlib::Absolutepath              $path     = '/etc/modules-load.d',
+  Optional[String]                  $content  = undef,
+  Optional[String]                  $source   = undef,
```

Comment:
```suggestion
  Optional[String[1]]               $source   = undef,
```
do we need to allow empty strings?

Reply:
empty strings are useful since that is I actual use case where I want to clobber a broken package config - https://bugzilla.redhat.com/show_bug.cgi?id=2044789

Having said that forcing a comment into these files is a good idea..